pubsublite/internal/wire/streams.go

```diff
@@ -63,11 +65,15 @@ type streamHandler interface {
 	validateInitialResponse(interface{}) error
 
 	// onStreamStatusChange is used to notify stream handlers when the stream has
-	// changed state. A `streamReconnecting` status change is fired before
-	// attempting to connect a new stream. A `streamConnected` status change is
-	// fired when the stream is successfully connected. These are followed by
-	// onResponse() calls when responses are received from the server. These
-	// events are guaranteed to occur in this order.
+	// changed state.
+	// - A `streamReconnecting` status change is fired before attempting to
+	//   connect a new stream.
+	// - A `streamResetState` status change may be fired if the server broke a
```

Comment:
It feels a bit strange to have streamResetState fire after streamReconnecting. Shouldn't we have the opposite order?

Reply:
Ah yes, that's the order I initially had, and then ran into a couple of issues:

* The receivers stop/disable things upon `streamReconnecting` and also need to do the same actions for `streamResetState`, so it was more practical to send the former first to avoid the duplication.
* When the stream is reinitialized, the server can send the `reset` signal straight after the initial request. So we should fire these notifications consistently in this order.

Amended the comments to hopefully reduce the confusion.